pyvrp/cpp/search/Route.h

```diff
@@ -985,14 +1013,14 @@ std::pair<Duration, Duration> Route::Proposal<Segments...>::duration() const
         auto ds = segment.duration(profile);
         auto first = segment.first();
 
-        if (first < data.numDepots())  // segment starts at depot
+        if (segment.startsAtReloadDepot())
             ds = ds.finaliseFront();
 
         auto const merge = [&](auto const &self, auto &&other, auto &&...args)
         {
             auto edgeDur = matrix(other.last(), first);
 
-            if (other.last() < data.numDepots())  // other ends at a depot
+            if (other.endsAtReloadDepot())
             {
                 // We can only finalise the current segment at the depot, so we
                 // need to travel there. The depot time windows restrictions are
```

Comment:
This returns us to the depot to finalise, but we should not do that if other is just the start depot: doing so removes all release time, which we need at the start depot to account for time warp due to depot or vehicle time windows interacting with the release time.

So we should only finalise here if we're at a *reload* depot, not at the start/end depots.